Lib/_osx_support.py

```diff
@@ -331,6 +355,12 @@ def compiler_fixup(compiler_so, cc_args):
             except ValueError:
                 break
 
+    elif not _supports_arm64_builds():
+        # Look for "-arch arm64" and drop that
+        for idx in range(len(compiler_so)):
```

Comment:
When `idx` is `len(compiler_so)-1`, it will throw exception.
So, it should be `for idx in range(len(compiler_so)-1):`

Reply:
dang, looks like you're right

Reply:
This PR was merged over a year ago and has long since been released to the field. Please open a new issue on the bug tracker and, if possible, provide a PR.  Otherwise, it will likely be forgotten about it. Thanks!